src/components/Table/sortRows.ts

```diff
@@ -27,21 +27,47 @@ function sortBatch<R extends Kinded>(
   caseSensitive: boolean,
 ): GridDataRow<R>[] {
   // When client-side sort, the sort value is the column index
-  const [value, direction] = sortState;
+  const [value, direction, persistent] = sortState;
+  
   const column = columns[value];
   const invert = direction === "DESC";
 
+  let persistentColumn: GridColumn<R, {}>;
+  if (persistent) {
+    persistentColumn = columns[persistent]
+  }
+
   // Make a shallow copy for sorting to avoid mutating the original list
   return [...batch].sort((a, b) => {
     const v1 = sortValue(applyRowFn(column, a, {} as any, 0), caseSensitive);
     const v2 = sortValue(applyRowFn(column, b, {} as any, 0), caseSensitive);
     const v1e = v1 === null || v1 === undefined;
     const v2e = v2 === null || v2 === undefined;
+
+    let p1 = false;
+    let p2 = false;
+    if (persistentColumn) {
+      p1 = sortValue(applyRowFn(persistentColumn, a, {} as any, 0), caseSensitive);
+      p2 = sortValue(applyRowFn(persistentColumn, b, {} as any, 0), caseSensitive);
+    }
     if ((a.pin || b.pin) && !(a.pin === b.pin)) {
       const ap = a.pin === "first" ? -1 : a.pin === "last" ? 1 : 0;
       const bp = b.pin === "first" ? -1 : b.pin === "last" ? 1 : 0;
       return ap === bp ? 0 : ap < bp ? -1 : 1;
-    } else if ((v1e && v2e) || v1 === v2) {
+    } else if (p1 && p2){
```

Comment:
Suggestion: This is checking truthiness, but if one row is a favorite (p1=1/true) and the other row is now (p2=0/false), then we probably still want to go into this loop and tell the sort that row 1 should come before row 2?

I.e. seems like this should maybe use `else if (persistentColumn)`, like any time we have a persistent sort in play, we should compare p1 and p2 of both rows.

Not just if they're both set, or both truthy.

Reply:
(Fwiw this is the sort of thing where, unless I'm missing something, I'm missing that it looks like sorting is working on the test data that we happen to be testing with, but with various random combinations, it is actually not fully working.)

Reply:
nod nod,

below this statement we have 

```
    else if (p1) {
      return -1; 
    } else if (p2) {
      return 1;
    }
```
    
in the case that only one of the persistent values is `true` it will always be sorted above a non persistent value. Furthermore I believe this is also the location to modify if we would like the ability to support sorting for persistent values to the top or bottom of the list .

as for test cases I think there can definitely be more and will try to address these corner cases. 

This is however effectively demonstrated in the `GridTable.stories.tsx:favoritingMore` ..... .terrible name